packages/pwa-kit-extension-sdk/src/shared/classes/application-extension-base.ts

```diff
@@ -26,9 +26,30 @@ export abstract class ApplicationExtension<Config extends ApplicationExtensionCo
      * @param config - The configuration object used to set up the extension.
      */
     constructor(config: Config) {
+        const cls = this.constructor as typeof ApplicationExtension
+        if (!cls.id) {
+            // This is a long standing typescript issue that typescript cannot enforce
+            // that the static property is defined in the subclass. So we need to throw
+            // an error here to let the developer know that they need to define the
+            // static property in their subclass.
+            // We can remove this once TS supports abstract static properties.
+            throw new Error(
+                `Static readonly property 'id' must be defined in class ${cls.name}`
+            )
+        }
         this.config = config
     }
 
+    /**
+     * Returns the id of the extension.
+     *
+     * @protected
+     * @returns id - The id of the extension.
+     */
+    static getId(): string {
+        return this.id
+    }
```

Comment:
I'm going to remove this since the `id` property is readonly. I feel like we don't need a getter for a readonly property.